### PR TITLE
Remove CrateNum parameter for queries that only work on local crate

### PR DESCRIPTION
**Pull request by cjgillot**
The pervasive `CrateNum` parameter is a remnant of the multi-crate rustc idea.

Using `()` as query key in those cases avoids having to worry about the validity of the query key.

### CODE REVIEW

**Comment by marmeladema** [comment]
I have tried something similar in https://github.com/rust-lang/rust/pull/71648 but I gave up^^

**Comment by Xanewok** [comment]
> (...) remnant of the multi-crate rustc idea

Sorry for the noise here but is this idea abandoned/discussed recently somewhere?

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 1]
```diff
@@ -928,9 +928,7 @@ impl<'hir> intravisit::Map<'hir> for Map<'hir> {
     }
 }
 
-pub(super) fn index_hir<'tcx>(tcx: TyCtxt<'tcx>, cnum: CrateNum) -> &'tcx IndexedHir<'tcx> {
-    assert_eq!(cnum, LOCAL_CRATE);
-
+pub(super) fn index_hir<'tcx>(tcx: TyCtxt<'tcx>, (): ()) -> &'tcx IndexedHir<'tcx> {
```
Comment (varkor) [inline compiler/rustc_middle/src/hir/map/mod.rs:931]: Why have a `(): ()` parameter instead of simply removing it?

**Comment by varkor** [comment]
I think it'd be better for someone more familiar with the query system to review here.

r? @oli-obk

File: compiler/rustc_middle/src/hir/map/mod.rs (hunk 1 above)
Comment (cjgillot) [reply compiler/rustc_middle/src/hir/map/mod.rs:931]: The query system is designed to work with queries taking exactly two arguments: the TyCtxt and a key. I am not convinced that removing the extra `()` parameter is worth the added complexity in the query system. (I am not even sure this is possible without a sizeable modification to the query plumbing.)

**Comment by oli-obk** [comment]
> 
> 
> > (...) remnant of the multi-crate rustc idea
> 
> Sorry for the noise here but is this idea abandoned/discussed recently somewhere?

niko mentioned this in https://github.com/rust-lang/rust/pull/71648#issuecomment-620726234

considering the approval of niko and eddyb in https://github.com/rust-lang/rust/pull/71648 let's go ahead with this idea

the changes themselves lgtm

@bors r+ rollup=never (bitrotty I presume)

**Pull request merged**